subprojects/docs/src/docs/userguide/javaLibraryPlugin.xml

```diff
@@ -0,0 +1,276 @@
+                I.e. , <firstterm>public</firstterm>, <firstterm>protected</firstterm> and <firstterm>package private</firstterm> members in the Java world)
+            </listitem>
+            <listitem>types used in public fields</listitem>
+            <listitem>public annotation types</listitem>
+        </itemizedlist>
+        <para>In opposition, any type that is used in the following list is irrelevant to the ABI, and therefore should be declared as <literal>implementation</literal> dependency:
+        </para>
+        <itemizedlist>
+            <listitem>types exclusively used in method bodies</listitem>
+            <listitem>types exclusively used in private members</listitem>
+            <listitem>types exclusively found in internal classes (future versions of Gradle will let you declare which packages belong to the public API)</listitem>
+        </itemizedlist>
+        <para>In the following sample, we can make the difference between an API dependency and an implementation dependency:</para>
+        <sample id="useJavaLibraryPlugin" dir="java-library/quickstart" title="Imports found in a class">
+            <sourcefile file="src/main/java/org/gradle/HttpClientWrapper.java" snippet="imports"/>
+        </sample>
+        <para>We can see that our class imports third party classes, but imports alone won't tell us if a dependency is an API or implementation dependency.
+            For this, we need to look at the methods. The following method used <literal>HttpClient</literal> as a parameter, so it belongs to the public API:
+        </para>
+        <sample id="useJavaLibraryPlugin" dir="java-library/quickstart" title="A dependency type is used in the public API">
+            <sourcefile file="src/main/java/org/gradle/HttpClientWrapper.java" snippet="api-dependency"/>
+        </sample>
+        <para>On the other hand, the <literal>ExceptionUtils</literal> type, coming from the <literal>commons-lang</literal> library, is only used in a method
+            body, so it's an implementation dependency:
+        </para>
+        <sample id="useJavaLibraryPlugin" dir="java-library/quickstart" title="A dependency type is used in the method bodies only">
+            <sourcefile file="src/main/java/org/gradle/HttpClientWrapper.java" snippet="implementation-dependency"/>
+        </sample>
+        <para>Therefore, we can deduce that <literal>commons-httpclient</literal> is an API dependency, whereas <literal>commons-lang</literal> is an implementation
+            dependency, which directly translates into the build file:
+        </para>
+        <sample id="useJavaLibraryPlugin" dir="java-library/quickstart" title="Declaring API and implementation dependencies">
+            <sourcefile file="build.gradle" snippet="dependencies"/>
+        </sample>
+        <para>It is in general safer to assume that a dependency is an implementation dependency only: leakage of implementation types to consumers would
```

Comment:
What does "safe" mean? It's either API or it is not. We can of course recommend "Make an effort to keep your API surface small". But there is nothing "unsafe" about it I think.